File: 장주성/20230511/폰켓몬.py
def solution(nums):
    answer = 0
    counts = {}

    for num in nums:
        if num in counts:
            counts[num] += 1
        else:
            counts[num] = 1
    # 입력으로 주어진 배열을 폰켓몬 종류별로 분류하고, 폰켓몬 종류 별 폰켓몬 수를 구함

    half = len(nums) / 2
    # 최대로 가져갈 수 있는 N/2마리의 포켓몬 수

    if len(counts) >= half:
        answer = half
    else:
        answer = len(counts)
    # 가져갈 수 있는 폰켓몬 수가 N/2마리 이상이면 N/2마리 가져감
    # 그보다 적으면, 가져갈 수 있는 폰켓몬만 가져감

    return answer

File: 서예진A/20230511/폰켓몬.cpp
#include <vector>
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    /*
    번호 순대로 정렬 -> 중복 제거
    N/2마리 가져가야하므로 answer이 N/2보단 크면 안됨
    */
    
    int answer = 0;
    int nums_size = nums.size(); //원래 배열 사이즈
    
    sort(nums.begin(),nums.end()); // 정렬
    
    nums.erase(unique(nums.begin(),nums.end()),nums.end()); // 중복제거
    
    int num=nums.size(); //중복 제거한 배열 사이즈
    
    
    if(num>nums_size/2)
        answer=nums_size/2;
    else answer=num;
    //but, 중복 제거한 배열 사이즈 값이 N/2마리보다 큰 경우 N/2마리가 answer값
    
    return answer;
}